packages/backend/src/nest/storage/storage.service.ts

```diff
@@ -183,22 +184,21 @@ export class StorageService extends EventEmitter {
   }
 
   public async initDatabases() {
-    this.logger('1/3')
     console.time('Storage.initDatabases')
 
+    this.logger('1/3')
+    this.attachStoreListeners()
+
     // FIXME: This is sort of messy how we are initializing things.
     // Currently, the CommunityMetadataStore sends an event during
     // initialization which is picked up by the CertificatesStore, but
     // the CertificatesStore is not initialized yet. Perhaps we can
-    // attach `this` as an EventEmitter first and then load data.
-    await this.communityMetadataStore.init(this)
-    await this.certificatesStore.init(this)
-    await this.certificatesRequestsStore.init(this)
-    await this.userProfileStore.init(this)
-
+    // initialize stores first and then load data/send events.
     this.logger('2/3')
-    await this.attachCertificatesStoreListeners()
-    await this.attachCsrsStoreListeners()
+    await this.communityMetadataStore.init()
+    await this.certificatesStore.init()
+    await this.certificatesRequestsStore.init()
+    await this.userProfileStore.init()
```

Comment:
Stores emit their own events now. The other changes uncovered a race between initializing the stores and receiving their first events.